src/backend/mpi/swdsm.cpp

```diff
@@ -930,12 +876,8 @@ void argo_initialize(std::size_t argo_size, std::size_t cache_size){
 	tmpcache=globalSharers;
 	vm::map_memory(tmpcache, gwritersize, current_offset, PROT_READ|PROT_WRITE);
 
-	current_offset += gwritersize;
-	tmpcache=lockbuffer;
-	vm::map_memory(tmpcache, pagesize, current_offset, PROT_READ|PROT_WRITE);
-
 	if (dd::is_first_touch_policy()) {
-		current_offset += pagesize;
+		current_offset += gwritersize;
```

Comment:
this appears to be a change in behaviour. Is this a silent bugfix, or something that went wrong when refactoring?

Reply:
> this appears to be a change in behaviour. Is this a silent bugfix, or something that went wrong when refactoring?

This change is prompted by the removal of `lockbuffer` of size `pagesize` from the memory mappings.

If I understand it correctly, `current_offset` is increased by the size of the _previous_ mapping ahead of every new mapping (in the sense of code formatting), which is why this is a little bit unclear. Since the mapping of `lockbuffer` (removed) no longer happens, I moved `current_offset += gwritersize` to the next position (replacing `current_offset += pagesize` for lockbuffer) to avoid changing behaviour. Perhaps changing the behaviour alltogether to increase after each mapping would be more clear.

Do let me know if I have misunderstood anything!

Reply:
sounds correct